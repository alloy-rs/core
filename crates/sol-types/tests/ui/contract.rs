use alloy_sol_types::sol;

sol! {
    contract C {
        contract Nested {}
    }
}

sol! {
    interface C {
        library Nested {}
    }
}

sol! {
    abstract contract C {
        interface Nested {}
    }
}

fn main() {}
