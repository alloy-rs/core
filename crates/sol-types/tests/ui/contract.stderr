error: cannot declare nested contracts
 --> tests/ui/contract.rs:5:18
  |
5 |         contract Nested {}
  |                  ^^^^^^

error: cannot declare nested contracts
  --> tests/ui/contract.rs:11:17
   |
11 |         library Nested {}
   |                 ^^^^^^

error: cannot declare nested contracts
  --> tests/ui/contract.rs:17:19
   |
17 |         interface Nested {}
   |                   ^^^^^^
